pkg/channel/http/http_channel.go

```diff
@@ -235,24 +240,29 @@ func (h *Channel) invokeMethodV1(ctx context.Context, req *invokev1.InvokeMethod
 	diag.DefaultHTTPMonitoring.ClientRequestStarted(ctx, verb, req.Message().Method, int64(len(req.Message().Data.GetValue())))
 	startRequest := time.Now()
 
-	// Send request to user application
-	resp := fasthttp.AcquireResponse()
```

Comment:
Why is this removed?

Reply:
Sadly, this was a side-effect of converting from fasthttp.Request to fasthttp.RequestCtx,

One solution was simply getting one fasthttp.Response from the pool and set on the fasthttp.RequestCtx, but it's not possible because responses doesn't support copying, and fasthttp.RequestCtx requires a value rather than pointer,

I can use ctx.Response.CopyTo(&resp) but as per [documentation](https://pkg.go.dev/github.com/valyala/fasthttp#Response.CopyTo) it doesn't copy the body stream

<img width="664" alt="image" src="https://user-images.githubusercontent.com/5839364/189724864-66122a48-11c9-4876-b9c2-b9ceba97810f.png">

Also, if the idea is to provide response recycling and less GC pressure I think that copying won't work anyways,

Any suggestion on this @yaron2 ?

Reply:
I'm completely aligned if you think this is a blocker for this PR. Especially if there have been previous experiences that resulted in more memory usage and increased latency (related to GC pauses)

Reply:
Currently I'd put this as a blocker but it doesn't mean it'll remain that way unless fixed. We need to be able to measure the performance hit here. An immediate solution would be to revert to the current (and more performant) request construction if no middleware is present.

Reply:
Great idea @yaron2. Applied your suggestions

> We need to be able to measure the performance hit here

Yes, I'll probably ask `@tanvigour` for help to look at the graphs we already have that the memory consumption/p99 latency increases